src/todo.ts

```diff
@@ -0,0 +1,99 @@
+import { Item } from './item';
+class TodoApp {
+    /**
+     * Allows the creation and maintenance of a simple todo list
+     */
+
+    public toDoList: Map<number, Item>;
+    public curId: number;
+
+    constructor() {
+        this.curId = 1;
+        this.toDoList = new Map<number, Item>();
+    }
+
+    // Had to generate the item ID here so two todoApps could have separate id lists
+    public addItem(description: string, dueAt: string): Item {
+        const date = new Date(dueAt);
+        if (date.toString() === 'Invalid Date') {
+            throw new Error('The date you entered is invalid');
+        }
+        const item = new Item(description, date, this.curId++);
+        this.toDoList.set(item.id, item);
+        return item;
+    }
+    public checkItem(id: number): boolean {
+        if (!this.toDoList.has(id)) {
+            throw new Error('No such item exists');
+        }
+        this.toDoList.get(id).done = true;
+        return true;
```

Comment:
So why return a boolean here?  What is gained?  If it is successful then no exceptions occur.  Else they do.